src/renderer/library/components/settings/Settings.tsx

```diff
@@ -235,122 +237,175 @@ const SaveCreatorSettings: React.FC<{}> = () => {
     );
 };
 
-const ApiKeys = () => {
+const ApiKeyComponent = ({ k, index }: {k: IApiKey, index: number}) => {
```

Comment:
pass the array index in component props is generally not a good idea, you should copy the api key/provider/(id?) tuple instead